### ISSUE TITLE
with voiceover, hitting the voice message button freezes voiceover practically forever

### ISSUE THREAD

**Issue by derekriemer** [signalapp/Signal-iOS#2333]
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description

With voiceover on (Apples built in screen reader for the blind), if I double tap and hold the voice message button (Placed where send is if there is no text), a message is indeed recorded, but unfortunately, the whole screen reader locks up after the hold is released.

### Steps to reproduce

- either go to settings > general > accessibility > voiceover, and toggle it on, or tell siri "Enable voiceover." 
    Note: voiceover changes how you navigate the screen, because blind people cannot tap a button without activating it, and need a mode to explore the screen with. To use voiceover, you can simply double tap any button to actually activate it. First touch the button, then when it says it, double tap the screen anywhere, or keep that finger on the button, and flick the screen with another finger.
- Open signal.
- Tap a persons name (Remember, double tap).
- Touch the voice message button (right middle above p).
- Now, act like you are going to perform a double tap, but instead of releasing your finger on tap 2, hold. This performs a tap and hold (Which is how you do a voice message).
- Once held, speak your voice message.
- Now, release your finger.

**Actual result:** Voiceover locks up and even if I press the home button, it still freezes. Sometimes eventually I can get it back, but I haven't found a consistent way.
**Expected result:** Voiceover resumes talking.

### Device info

**Device**: IPhone 6
**iOS version**: 10.latest
**Signal version:** 2.13.4.0

### Link to debug log

https://gist.github.com/anonymous/9d17edf18b1fe39d688d5a563da2f1fa

**Comment by derekriemer** [comment]
Update: to get speech back, open sirri, and ask her to disable and reenable voiceover.

**Comment by MarcoZehe** [comment]
Confirmed. Sometimes Voice comes back if you toggle the mute switch on the side of the iPhone, but the Earcons VoiceOver plays as navigation help will remain muted. I suspect it has something to do with the audio route restauration after the voice message has finished recording. VoiceOver continues to function in principle, but the audio routes are messed up, so VoiceOver no longer knows which channel to speak through. iPhone 7, iOS 10.3.2.

**Comment by heull001** [comment]
This bug exists in iOS 11, too. It is important to fix it soon because many blind people use voice message instead of text messages.

**Comment by mrkiko** [comment]
Experimented something like this, yes.

**Comment by MarcoZehe** [comment]
This seems to have long been fixed. Or can you still reproduce this, @derekriemer ?
